pkg/api/validate/dynamic/dynamic.go

```diff
@@ -40,7 +41,7 @@ type Subnet struct {
 }
 
 type ServicePrincipalValidator interface {
-	ValidateServicePrincipal(ctx context.Context, clientID, clientSecret, tenantID string) error
+	ValidateServicePrincipal(ctx context.Context, clientID, clientSecret, tenantID string, azureClaimHelper azureclaim.AzureClaim, getServicePrincipalToken getTokenWrapper) error
```

Comment:
1. I see that we changed the intraface signature, but do not see that we updated the calling code (there are two places at the moment of writing) where we call a method implementing this interface. This is probably the reason why CI is faling.
2. Can we do that withouch changing the signature?

Reply:
@m1kola Thank you for pointing out that I wasn't updating all references to this method. As far as changing the signature, I think introducing that small layer of abstraction is the only way to test the function itself. The only other approach I can think of is re-creating the steps of the method in the test itself which in my opinion isn't the correct approach. I am happy to hear any suggestions - @bennerv @bryanro92